twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillRunnerService.java

```diff
@@ -211,7 +226,7 @@ public String getJVMOptions() {
   @Override
   public Cancellable scheduleSecureStoreUpdate(final SecureStoreUpdater updater,
                                                long initialDelay, long delay, TimeUnit unit) {
-    if (!UserGroupInformation.isSecurityEnabled()) {
+    if (!UserGroupInformation.isSecurityEnabled() || !enableSecureStoreUpdate) {
```

Comment:
You shouldn't disable this one. Otherwise user won't be able to schedule his own update. We should only disable the default one by Twill (the one that update HDFS delegation tokens in `startUp`).